cmd/ctr/commands/run/run_unix.go

```diff
@@ -78,6 +83,20 @@ func NewContainer(ctx gocontext.Context, client *containerd.Client, context *cli
 	if context.Bool("net-host") {
 		opts = append(opts, oci.WithHostNamespace(specs.NetworkNamespace), oci.WithHostHostsFile, oci.WithHostResolvconf)
 	}
+	joinNs := context.StringSlice("with-ns")
+	for _, ns := range joinNs {
+		parts := strings.Split(ns, ":")
+		if len(parts) != 2 {
+			return nil, errors.New("joining a Linux namespace using --with-ns requires the format 'nstype:path'")
```

Comment:
Actually, we just need path and can easily detect the type with the (readlink-ed) path string.

Reply:
not true

Reply:
Is there any corner case that makes it hard to detect the type?

Reply:
bind mounts, more than one ns per fd

Reply:
uh, ok

Reply:
"uh, ok"?  why do you think `setns` takes a type? readlink does not work on bind mounts of namespaces.

Reply:
I meant, "uh, ok, I understood readlink does not work with bind-mounted ns".
Sorry for causing confusion